### PR TITLE
Capstone-Project-2

### PR DESCRIPTION
**Pull request by AnsarIbrahim**
### This PR contains:

- Find an API so you can base the development of the web app around it ✅
- [TVmaze API] ✅
- Use Involvement API to record user interactions likes and comments ✅
- The home page ✅
- The comments popup ✅
- Follow the layout of the wireframes provided ✅
- When the page loads, the web app retrieves data ✅
- The selected API shows the list of items on the screen ✅
- The Involvement API to show the item likes ✅
- One to the base API and another one to the Involvement API ✅
- When the user clicks on the Like button of an item, the interaction is recorded in the Involvement API and the screen is 
updated ✅
- The Comments popup appears when the user clicks on the "Comments" button ✅
- Home page header and navigation similar to the given mockup ✅
- The home page footer is similar to the given mockup✅
- Comments popup ✅
- When the popup loads, the webapp retrieves data ✅
- The selected API shows details about the selected item ✅
- The Involvement API to show the item comments ✅
- When the user clicks on the "Comment" button, the data is recorded in the Involvement API, and the screen is updated ✅
- Counters We have counters in all the interfaces that show ✅
- The number of items (home) ✅
- The number of comments (comments popup) ✅
- Set up the repository on GitHub and use Gitflow ✅
- Set up webpack ✅
- Set up a JavaScript testing library (Jest) ✅
- A description of the project ✅
- A demo of the project features ✅
- Using Kanban Project ✅

### CODE REVIEW

File: src/js/module/test/comment.test.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    creation_date: '2023-06-28',
+    username: 'JohnDoe',
+    comment: 'This is a test comment.',
+  };
+
+  const expectedHTML = `<p>${comment.creation_date} ${comment.username}: ${comment.comment}</p>`;
+
+  const generatedHTML = generateCommentHTML(comment);
+
+  expect(generatedHTML.trim()).toBe(expectedHTML.trim());
```
Comment (thecodechaser) [inline src/js/module/test/comment.test.js]: - [ ] This comment popup test is failing, and for the comment test, you need to test the number of comments, not the HTML as requested in the project requirements. Please make sure to test the correct values and the test should pass without any issues.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @YoutLone, and @AnsarIbrahim👋, 

### You are doing a marvelous job on the JavaScript group capstone project and this is one of the best projects I have seen so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Nice UI design. ✔️
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Highly organized code. ✔️
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️

- [ ] Please make sure the maximum length of presentation video is 5 minutes as requested in the project requirements. Currently, the video length is 5:08 minutes. You should trim the video and make it a maximum of 5 minutes lengthy.

- [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/js/module/test/comment.test.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    username: 'JohnDoe',
+    comment: 'This is a test comment.',
+  };
+
+  const expectedHTML = `<p class="mt-2 comment-p">${comment.creation_date} ${comment.username}: ${comment.comment}</p>`;
+
+  const generatedHTML = generateCommentHTML(comment);
+
+  expect(generatedHTML.trim()).toBe(expectedHTML.trim());
+});
```
Comment (anagudelogu) [inline src/js/module/test/comment.test.js:15]: - [ ] You did a great job working on this milestone! 💯 However, according to the previous reviewer, please make sure that you are testing the `Comments counter` feature as stated in the project requirements. At the moment, you're correctly testing that your `generateCommentHTML` is working, but this doesn't give us enough confidence to affirm the same for the counter. Please consider adding this test case! 🤓 🔧

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 🚧 

Hi @YoutLone 👋🏼 
I'm Andres.

You've done a great job implementing some of the requested changes for this milestone 🚀, but there are still some which haven't been addressed yet. You're getting closer, keep up the good work! 🤖 🎆 



 ## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Great UI design. 💻 
✅ Descriptive PR title, summary and commit messages. 💬 


 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ghost — APPROVED** [review]
Hi @AnsarIbrahim , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Correct use of Gitflow
✅ No linter error
✅ App is well tested 

## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**